src/utils/media-utils.js

```diff
@@ -113,6 +117,18 @@ export const determineMediaDuration = memoizee(async url => {
   return Promise.race([playerPromise, timeoutPromise]);
 }, { promise: true });
 
+export const verifyYoutubeThumbnailUrl = memoizee(url => {
```

Comment:
Maybe `verifyMediaUrl` (due to the use of `MEDIA_TIMEOUT_IN_MS`) would be better? Since this method doesn't know anything of youtube actually?

Reply:
I went with `verifyMediaThumbnailUrl` to not suggest it would be video or audio.